packit/api.py

```diff
@@ -362,6 +363,129 @@ def update_source_git(self, revision_range: str):
+
+    def _get_latest_commit_update_pair(self) -> Tuple[str, str]:
+        """Finds the latest pair of commits which was created by updating
+        source-git from dist-git (or vice versa) denoted by git trailers.
+
+        Determining the latest pair by commit time is not sufficient since
+        the datetime may be the same in all commits. We need to check for
+        actual parent-children relationships to determine the ordering.
+
+        Returns:
+            A tuple containing hash of the source-git commit and hash of the
+                corresponding dist-git commit. This represents the latest
+                update pair in both source-git and dist-git.
+
+        Raises:
+            PackitException, if no commits with git trailers that are used
+                for checking the sync could be found.
+        """
+        # Use source-git commit first in both tuples for consistency
+        sg_update_commits = [
+            (
+                c.hexsha,
+                re.search(rf"{re.escape(FROM_DIST_GIT_TOKEN)}: (.+)", c.message).group(
+                    1
+                ),
+            )
+            for c in self.up.local_project.get_commits(
+                max_count=1, grep=rf"^{re.escape(FROM_DIST_GIT_TOKEN)}: .\+$"
+            )
+        ]
+        dg_update_commits = [
+            (
+                re.search(
+                    rf"{re.escape(FROM_SOURCE_GIT_TOKEN)}: (.+)", c.message
+                ).group(1),
+                c.hexsha,
+            )
+            for c in self.up.local_project.get_commits(
+                max_count=1, grep=rf"^{re.escape(FROM_SOURCE_GIT_TOKEN)}: .\+$"
+            )
+        ]
+        if sg_update_commits and dg_update_commits:
+            # Check ancestor relationships, the situation is as follows:
+            #     source-git HEAD          dist-git HEAD
+            #           |                       |
+            # sg_update_commits[0][0] <- sg_update_commits[0][1]
+            #           |                       |
+            # dg_update_commits[0][0] -> dg_update_commits[0][1]
+
+            # sg_update_commits and dg_update_commits could be swapped,
+            # we need to determine the correct order using
+            #   git merge-base --is-ancestor
+            # Ancestor is the older commit, we want to return the newer
+            if self.up.local_project.git_repo.is_ancestor(
+                sg_update_commits[0][0], dg_update_commits[0][0]
+            ):
+                return dg_update_commits[0]
+            else:
+                return sg_update_commits[0]
+        elif sg_update_commits:
+            return sg_update_commits[0]
+        elif dg_update_commits:
+            return dg_update_commits[0]
+        else:
+            raise PackitException(
+                "No git commits with trailers to mark synchronization points were found."
+            )
+
+    def sync_status(self) -> None:
```

Comment:
Could this method be written so that it will be possible to use it to check the sync-status of the repositories when running `update-dist-git` and `update-source-git`? Or if not this one, then maybe the `_show_status` one?

Reply:
Hm, good idea. I am thinking what the most convenient way to approach this is. Perhaps the `sync_status` could just return the commit ranges (either as a tuple of two strings (or None if up-to-date) or it could be thrown into a dataclass to represent the sync status) and not call `_show_status`. I'd then convert `_show_status` to a public method taking the ranges from `sync_status` and the CLI method would call both `sync_status` and `show_status`. This would make it also more testable as I wouldn't have to rely on the mock. Does this sound good to you?

Reply:
Sounds good :+1: